partner_event/models/event_registration.py

```diff
@@ -34,16 +35,21 @@ def create(self, vals):
             Event = self.env['event.event']
             # Look for a partner with that email
             email = vals.get('email').replace('%', '').replace('_', '\\_')
+            event = Event.browse(vals['event_id'])
+            company_id = event.company_id.id
             attendee_partner = Partner.search([
-                ('email', '=ilike', email)
+                ('email', '=ilike', email),
+                '|',
+                ('company_id', '=', company_id),
+                ('company_id', '=', False),
             ], limit=1)
-            event = Event.browse(vals['event_id'])
             if attendee_partner:
                 vals['name'] = vals.setdefault('name', attendee_partner.name)
                 vals['phone'] = vals.setdefault(
                     'phone', attendee_partner.phone)
             elif event.create_partner:
                 # Create partner
+                vals['company_id'] = company_id
                 attendee_partner = Partner.sudo().create(
                     self._prepare_partner(vals))
```

Comment:
This is unrelated, but isn't this going to generate warning due to invalid fields that are on `event.registration`, but not on `res.partner`?